app/assets/javascripts/battletype/dockyard.js

```diff
@@ -6,20 +6,21 @@
   this.Dockyard = {
     ps2Port: undefined,
     _templates: {},
-    
+
     launch: function (attributes, combatZone) {
-      var newShip = Ship.build(this._templates[attributes.ship.type], attributes);
-      
+      var newShip   = Ship.build(this._templates[attributes.ship.type], attributes);
+      var positionY = combatZone.randomFreeVerticalSlot - newShip.height;
+
       newShip.addEventListener("animationend", this._bombingEventDispatcherFor(newShip), false);
-      newShip.positionY = combatZone.randomFreeVerticalSlot;
+      newShip.positionY = Math.max(0, positionY);
```

Comment:
Since this line's role is to _assign_ a new value to `newShip.positionY`, I feel like this isn't the right place for _also_ guarding against a possible negative value of the `positionY` variable. We should either ensure that the value is correct _before_ assigning it, or making so that the `newShip.positionY` setter can handle negative values.

(I'd go with the latter, since I prefer patient objects that silently ignore invalid messages that I send them to fragile objects that make me check everything before I talk to them, or bitter objects that check the validity of the message they receive, but only to raise and throw if something's wrong.)

(Side effect of moving this logic to the `newShip` object: we could stop asking the `newShip` for its height, line 12, since the ship itself would handle that for us.)